backend/src/main/repositories/AccountRepository.ts

```diff
@@ -1,77 +1,81 @@
-// Temporary example of DAO implementation. We gonna implement an actual database later.
-// Repository class are for database manipulation only. Leave data logic for service layer.
-
 import debug from 'debug';
 import { injectable } from 'tsyringe';
-import { AccountCreationDTO } from '../dto/AccountCreationDTO';
+import { AccountCreationDTO, AccountUpdateDTO } from '../dto/Accounts/AccountDTOs';
 import { CRUD } from './CRUDInterface';
-const log: debug.IDebugger = debug('app:userReposity-example');
+const log: debug.IDebugger = debug('app:AccountRepository');
 import { Account } from '../models/Account';
 
 @injectable()
-// Repository class implements CRUD interface to force it to have at least the 4 basic crud operations
 export default class AccountRepository implements CRUD {
-
   constructor() {
-    log('Created new instance of AccountDao');
+    log('Created new instance of AccountRepository');
   }
 
   public create = async (accountInfo: AccountCreationDTO): Promise<Account> => {
-
-    log('added new user');
-    // Good practice to return the entity being manipulated. In normal case it would be the entity from the database.
-
-    const createdAccount = Account.build({ email: accountInfo?.email, firstName: "Bob", lastName: "Bobby", phoneNumber: "514-123-1234", username: "bob123", password: "ENCRYPTED PASSWORD" })
-    createdAccount.save();
-
-    return Promise.resolve(createdAccount);
+    try {
+      const createdAccount = Account.build(accountInfo);
+      createdAccount.save();
+
+      log(`Added new account ${createdAccount.email}`);
+      return Promise.resolve(createdAccount);
+    } catch (err: any) {
+      log(err);
+      return Promise.reject(err);
+    }
   };
 
-  public delete = async (email: string): Promise<string> => {
-    log(`Account with email ${email} has been deleted`);
-    
-    Account.destroy(
-      { where: { email: email } }
-    )
-
-    return Promise.resolve('A deleted entity');
+  public delete = async (email: string): Promise<number> => {
```

Comment:
Does the function destroy return an object? Shouldn't the return type be an object then?

Reply:
Never mind, just searched online and found that destroy and update return an integer that indicates the number of rows that were affected.